File: spring-boot-kafka-demo/kafka-simple-demo1/src/main/java/com/github/jccode/springbootkafkademo/simpledemo1/Application.java
package com.github.jccode.springbootkafkademo.simpledemo1;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.kafka.core.KafkaTemplate;

import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

@SpringBootApplication
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args).close();
    }



    private static final String TOPIC = "TEST_TOPIC";

    private static final Logger logger = LoggerFactory.getLogger(Application.class);

    private CountDownLatch latch = new CountDownLatch(3);

    @Autowired
    private KafkaTemplate<String, String> template;

    @Bean
    public CommandLineRunner commandLineRunner(ApplicationContext context) {
        return args -> {
            template.send(TOPIC, "hello");
            template.send(TOPIC, "world");
            template.send(TOPIC, "kitty");

            latch.await(60, TimeUnit.SECONDS);
            logger.info("All received!");
        };
    }

    @KafkaListener(topics = TOPIC)
    public void listen(ConsumerRecord<?, ?> cr) {
        logger.info(cr.toString());
        latch.countDown();
    }
}

File: spring-boot-kafka-demo/kafka-simple-demo1/src/main/scala/com/github/jccode/springbootkafkademo/simpledemo1/Application.scala
package com.github.jccode.springbootkafkademo.simpledemo1

import java.util.concurrent.{CountDownLatch, TimeUnit}

import org.apache.kafka.clients.consumer.ConsumerRecord
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.{CommandLineRunner, SpringApplication}
import org.springframework.context.annotation.Bean
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.kafka.core.KafkaTemplate

/**
  * Application
  *
  * @author 01372461
  */
@SpringBootApplication
class Application {
  private final val TOPIC = "TEST_TOPIC"
  private final val log: Logger = LoggerFactory.getLogger(classOf[Application])
  private val latch = new CountDownLatch(3)

  @Autowired val template: KafkaTemplate[String, String] = null

  @Bean
  def commandLineRunner = new CommandLineRunner {
    override def run(strings: String*): Unit = {
      template.send(TOPIC, "hello")
      template.send(TOPIC, "world")
      template.send(TOPIC, "kitty")

      latch.await(60, TimeUnit.SECONDS)
      log.info("All received!")
    }
  }

  @KafkaListener(topics = Array(TOPIC))
  def listen(cr: ConsumerRecord[_, _]): Unit = {
    log.info(cr.toString)
    latch.countDown()
  }

}

object Application {
  def main(args: Array[String]): Unit = {
    SpringApplication.run(classOf[Application], args: _*)
  }
}
